etna/models/nn/deepar_new/deepar.py

```diff
@@ -104,13 +117,19 @@ def forward(self, x: DeepARBatchNew, *args, **kwargs):  # type: ignore
         decoder_real = x["decoder_real"].float()  # (batch_size, decoder_length, input_size)
         decoder_target = x["decoder_target"].float()  # (batch_size, decoder_length, 1)
         decoder_length = decoder_real.shape[1]
-        weights = x["weight"]
+        weights = x["weight"]  # (batch_size)
+        segment_ids = x["segment_idx"]  # (batch_size)
+        encoder_embeddings = (
+            self.embedding(segment_ids).unsqueeze(1).expand([-1, encoder_real.shape[1], -1])
+        )  # (batch_size, encoder_length-1, hidden_size)
+        decoder_embeddings = self.embedding(segment_ids).unsqueeze(1)  # (batch_size, 1, hidden_size)
         forecasts = torch.zeros((decoder_target.shape[0], decoder_target.shape[1], self.n_samples))
-
         for j in range(self.n_samples):
-            _, (h_n, c_n) = self.rnn(encoder_real)
+            _, (h_n, c_n) = self.rnn(torch.cat((encoder_real, encoder_embeddings), dim=2))
```

Comment:
You should concat embedding to features only once, don't do it on each iteration!!!!